openml/datasets/functions.py

```diff
@@ -501,34 +496,26 @@ def create_dataset(name, description, creator, contributor,
 
     data = data.values if hasattr(data, "columns") else data
 
-    if format is not None:
-        warn("The format parameter will be deprecated in the future,"
-             " the method will determine the format of the ARFF "
-             "based on the given data.", DeprecationWarning)
-        d_format = format
-
-    # Determine ARFF format from the dataset
-    else:
-        if isinstance(data, (list, np.ndarray)):
-            if isinstance(data[0], (list, np.ndarray)):
-                d_format = 'arff'
-            elif isinstance(data[0], dict):
-                d_format = 'sparse_arff'
-            else:
-                raise ValueError(
-                    'When giving a list or a numpy.ndarray, '
-                    'they should contain a list/ numpy.ndarray '
-                    'for dense data or a dictionary for sparse '
-                    'data. Got {!r} instead.'
-                    .format(data[0])
-                )
-        elif isinstance(data, coo_matrix):
-            d_format = 'sparse_arff'
+    if isinstance(data, (list, np.ndarray)):
```

Comment:
Hm, don't you need to ckeck for `pd.DataFrame` and `pd.SparseDataFrame`, too?

Reply:
True